pkg/upgrade/upgrade.go

```diff
@@ -368,41 +370,49 @@ func UpdateFromLegacyVersion(cli client.Client, platform deploy.Platform, appNS
 
 		deprecatedServices := []string{"rhods-model-monitoring"}
-		multiErr = multierror.Append(multiErr, deleteDeprecatedResources(ctx, cli, montNamespace, deprecatedServices, &corev1.ServiceList{}))
+		multiErr = multierror.Append(multiErr, deleteDeprecatedResources(ctx, cli, dscMonitoringNamespace, deprecatedServices, &corev1.ServiceList{}))
 
 		deprecatedRoutes := []string{"rhods-model-monitoring"}
-		multiErr = multierror.Append(multiErr, deleteDeprecatedResources(ctx, cli, montNamespace, deprecatedRoutes, &routev1.RouteList{}))
+		multiErr = multierror.Append(multiErr, deleteDeprecatedResources(ctx, cli, dscMonitoringNamespace, deprecatedRoutes, &routev1.RouteList{}))
 
 		deprecatedSecrets := []string{"rhods-monitoring-oauth-config"}
-		multiErr = multierror.Append(multiErr, deleteDeprecatedResources(ctx, cli, montNamespace, deprecatedSecrets, &corev1.SecretList{}))
+		multiErr = multierror.Append(multiErr, deleteDeprecatedResources(ctx, cli, dscMonitoringNamespace, deprecatedSecrets, &corev1.SecretList{}))
 
 		deprecatedClusterroles := []string{"rhods-namespace-read", "rhods-prometheus-operator"}
-		multiErr = multierror.Append(multiErr, deleteDeprecatedResources(ctx, cli, montNamespace, deprecatedClusterroles, &authv1.ClusterRoleList{}))
+		multiErr = multierror.Append(multiErr, deleteDeprecatedResources(ctx, cli, dscMonitoringNamespace, deprecatedClusterroles, &authv1.ClusterRoleList{}))
 
 		deprecatedClusterrolebindings := []string{"rhods-namespace-read", "rhods-prometheus-operator"}
-		multiErr = multierror.Append(multiErr, deleteDeprecatedResources(ctx, cli, montNamespace, deprecatedClusterrolebindings, &authv1.ClusterRoleBindingList{}))
+		multiErr = multierror.Append(multiErr, deleteDeprecatedResources(ctx, cli, dscMonitoringNamespace, deprecatedClusterrolebindings, &authv1.ClusterRoleBindingList{}))
 
 		deprecatedServiceAccounts := []string{"rhods-prometheus-operator"}
-		multiErr = multierror.Append(multiErr, deleteDeprecatedResources(ctx, cli, montNamespace, deprecatedServiceAccounts, &corev1.ServiceAccountList{}))
+		multiErr = multierror.Append(multiErr, deleteDeprecatedResources(ctx, cli, dscMonitoringNamespace, deprecatedServiceAccounts, &corev1.ServiceAccountList{}))
 
 		deprecatedServicemonitors := []string{"modelmesh-federated-metrics"}
-		multiErr = multierror.Append(multiErr, deleteDeprecatedServiceMonitors(ctx, cli, montNamespace, deprecatedServicemonitors))
+		multiErr = multierror.Append(multiErr, deleteDeprecatedServiceMonitors(ctx, cli, dscMonitoringNamespace, deprecatedServicemonitors))
 	}
 
+	// Handling for dashboard Jupyterhub CR, see jira #443
+	JupyterhubApp := schema.GroupVersionKind{
+		Group:   "dashboard.opendatahub.io",
+		Version: "v1",
+		Kind:    "odhapplications",
+	}
+	multiErr = multierror.Append(multiErr, removOdhApplicationsCR(ctx, cli, JupyterhubApp, "jupyterhub", dscApplicationsNamespace))
```

Comment:
What is this deleting, file wise? I'm having a hard time reading Go here -- is https://github.com/opendatahub-io/odh-dashboard/blob/main/manifests/apps/jupyter/jupyterhub-app.yaml the resource you're deleting?

Reply:
mmmm I think this is still needed, idkw it's still called jupyterhub (the file) cause the name of the odhapplication is jupyter

Reply:
i do not follow this @lucferbux 
which resource you really want to delete from cluster:   the one originally generated from https://github.com/opendatahub-io/odh-dashboard/blob/main/manifests/apps/jupyter/jupyter-app.yaml?  named  "jupyter" 
or the one from  https://github.com/opendatahub-io/odh-dashboard/blob/main/manifests/apps/jupyter/jupyterhub-app.yaml, named  "juypterhub"

Reply:
This is part of https://github.com/opendatahub-io/odh-dashboard/pull/2513 -- we are not merging this sprint. So sorry @zdtsw to keep doing this to you but it has been chaotic the past few sprints. We will merge it next sprint at the start to give proper runway to test and find issues. The day before Code Freeze is a bad day to try and make this happen.

Can you linger this PR for 1 more week?

Reply:
sure, no worries. we can move it to 2.9

Reply:
merge this PR, since https://github.com/opendatahub-io/odh-dashboard/pull/2513 has been merged in dashboard